src/components/UserSelector.tsx

```diff
@@ -2,15 +1,68 @@
+import React, { useEffect, useRef, useState } from 'react';
+import { User } from '../types/User';
+import cl from 'classnames';
+import { pause } from '../utils/methods';
+import { Post } from '../types/Post';
+
+interface Props {
+  users: User[];
+  onActiveUser: (v: User) => void;
+  activeUser: User | null;
+  onShowSideBar: (v: boolean) => void;
+  onActivePost: (v: Post | null) => void;
+}
+
+export const UserSelector: React.FC<Props> = ({
+  users,
+  onActiveUser,
+  activeUser,
+  onShowSideBar,
+  onActivePost,
+}) => {
+  const [showDropDown, setShowDropDown] = useState(false);
+  const divRef = useRef<HTMLDivElement | null>(null);
+
+  useEffect(() => {
+    const handleClickedOutside = (e: MouseEvent) => {
+      if (divRef.current && !divRef.current.contains(e.target as Node)) {
+        setShowDropDown(false);
+      }
+    };
+
+    document.addEventListener('click', handleClickedOutside);
+
+    return () => {
+      document.removeEventListener('click', handleClickedOutside);
+    };
+  }, []);
+
+  const handleSelectUser = async (user: User) => {
+    onActiveUser(user);
+    setShowDropDown(false);
+    onShowSideBar(false);
+    onActivePost(null);
+    await pause();
+  };
 
-export const UserSelector: React.FC = () => {
   return (
-    <div data-cy="UserSelector" className="dropdown is-active">
-      <div className="dropdown-trigger">
+    <div
+      ref={divRef}
+      data-cy="UserSelector"
+      className={cl('dropdown', {
+        'is-active': showDropDown,
+      })}
+    >
+      <div
+        className="dropdown-trigger"
+        onClick={() => setShowDropDown(!showDropDown)}
```

Comment:
Ensure that the dropdown is accessible, including support for keyboard navigation and screen readers. This might involve handling keyboard events and providing appropriate ARIA attributes.